README.markdown

```diff
@@ -14,8 +14,7 @@ First [install **Racket** (v6.11 or later)](http://racket-lang.org), then
     $ git clone http://github.com/arclanguage/anarki
     $ cd anarki
     $ ./arc.sh    # start the interactive repl
-    $cd apps/news
-    $ ./run-news  # or start News (also with the repl)
+    $ (app-start "news")  # or start News (also with the repl)
```

Comment:
This isn't a shell command, is it?

There's currently no way to pass an expression to evaluate to `arc.sh` on the commandline.

Maybe instead of `app-start` being an arc function, it should be a shell-script that can be called like this?

```
./app-start news
```

Or even:
```
./arc.sh app news
```

What do you think?

Reply:
> There's currently no way to pass an expression to evaluate to arc.sh on the commandline.

I've added that feature [here](https://notabug.org/arc/knark/commit/4e0cd7fecff3be6387f1fa2e8ea51ed163749764). Can probably port it to **Anarki**.

Reply:
It works from arc.cmd in Windows so I assumed it would work through arc.sh as well. I was hoping to be able to get one solution that worked for both?

Reply:
Yes. I can run ./arc.cmd and then (app-start "news") from the prompt and it works fine.

Reply:
To be clear, that works in Unix as well.

Here, the `$` at the start indicates that you're at the shell prompt. What in Windows would be `c:\> ` or something like that. The Arc prompt is denoted by `arc> `. So maybe you just need to change the prompt above.

But as @hjek points out, it's nice to be able to start news or other apps in a single command, right from the shell prompt, without needing to type more at the Arc prompt.

No matter, make your change and one of us can switch it later.

Reply:
ok, I reverted the readme, so I guess we can keep the existing scripts for news.

Reply:
Cool, that works.

Can you just ensure that it continues to work after this change? I think you may need to make some change to it. Particularly after reverting the news.arc change.

Reply:
To be honest, since the point of this was to get rid of those scripts to begin with, I think it would be better to just not merge it. Redirecting srvdir* is still going to be an issue with new apps but that can be dealt with some other way, maybe.

Reply:
That's too bad. I still think the idea of a convention for arbitrary apps is a good one. And it'll need a driver function. Ideally you want to do most of the work in arc and add a tiny shim to the shell script.

See how arc.sh adds commands to run:

https://github.com/arclanguage/anarki/blob/77f93ee883/arc.sh#L120.

Can you just run `(app-start "news")` similarly from `run-news`?

Reply:
I might be able to make it work.